p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -847,42 +847,47 @@ public boolean hasAlreadyRemovedAddOncePayload(ProtectedStoragePayload protected
 
-        if (storedData == null) {
-            log.debug("We don't have data for that refresh message in our map. That is expected if we missed the data publishing.");
+            if (storedData == null) {
+                log.debug("We don't have data for that refresh message in our map. That is expected if we missed the data publishing.");
 
-            return false;
-        }
+                return false;
+            }
 
-        ProtectedStorageEntry storedEntry = map.get(hashOfPayload);
-        ProtectedStorageEntry updatedEntry = new ProtectedStorageEntry(
-                storedEntry.getProtectedStoragePayload(),
-                storedEntry.getOwnerPubKey(),
-                refreshTTLMessage.getSequenceNumber(),
-                refreshTTLMessage.getSignature(),
-                this.clock);
+            ProtectedStorageEntry storedEntry = map.get(hashOfPayload);
+            ProtectedStorageEntry updatedEntry = new ProtectedStorageEntry(
+                    storedEntry.getProtectedStoragePayload(),
+                    storedEntry.getOwnerPubKey(),
+                    refreshTTLMessage.getSequenceNumber(),
+                    refreshTTLMessage.getSignature(),
+                    this.clock);
 
 
-        // If we have seen a more recent operation for this payload, we ignore the current one
-        if (!hasSequenceNrIncreased(updatedEntry.getSequenceNumber(), hashOfPayload))
-            return false;
+            // If we have seen a more recent operation for this payload, we ignore the current one
+            if (!hasSequenceNrIncreased(updatedEntry.getSequenceNumber(), hashOfPayload))
+                return false;
 
-        // Verify the updated ProtectedStorageEntry is well formed and valid for update
-        if (!updatedEntry.isValidForAddOperation())
-            return false;
+            // Verify the updated ProtectedStorageEntry is well formed and valid for update
+            if (!updatedEntry.isValidForAddOperation())
+                return false;
 
-        // Update the hash map with the updated entry
-        map.put(hashOfPayload, updatedEntry);
+            // Update the hash map with the updated entry
+            map.put(hashOfPayload, updatedEntry);
 
-        // Record the latest sequence number and persist it
-        sequenceNumberMap.put(hashOfPayload, new MapValue(updatedEntry.getSequenceNumber(), this.clock.millis()));
-        requestPersistence();
+            // Record the latest sequence number and persist it
+            sequenceNumberMap.put(hashOfPayload, new MapValue(updatedEntry.getSequenceNumber(), this.clock.millis()));
+            requestPersistence();
 
-        // Always broadcast refreshes
-        broadcaster.broadcast(refreshTTLMessage, sender);
+            // Always broadcast refreshes
+            broadcaster.broadcast(refreshTTLMessage, sender);
 
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
```

Comment:
Maybe we should add an error log to give the stacktrace more context in the log file?